### PR TITLE
Switch to Photon for thumbnail generation

### LINKED ISSUE #979: Explore the use of Photon as a thumbnail service
## Description

We are presently using a version of [`imaginary`](https://github.com/h2non/imaginary) with a [custom build process for addressing a few open issues](https://github.com/WordPress/openverse-api/blob/87847eb0da81c3fb66a7dc7361e921fd02944f76/.github/workflows/build_imaginary.yml). The build process for this service is infrequent, but can take [several hours to run](https://github.com/WordPress/openverse-api/actions/workflows/build_imaginary.yml).

WordPress.com offers an open source image acceleration service called [Photon](https://developer.wordpress.com/docs/photon/) which I believe performs all of the requisite actions we would need in order to generate thumbnails for Openverse.

This would allow us to remove one of the services required to run this stack in production. We could also use the service itself locally, or [spin up a dockerized version of Photon for dev use](https://github.com/chriszarate/docker-photon).

There may be other reasons which might make Photon unsuitable for us, but it's worth exploring as a potential alternative in order to reduce our infrastructure complexity!

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #979 by @AetherUnbound

## Description

Now that the WordPress.com Photon instance supports an authentication header to allow us to forward query params, we're able to use it for thumbnail generation.

Note: This PR is still missing:

1. Add/move thumbnail unit tests to match new location `catalog.api.utils.photon` module

## Testing Instructions

Checkout the branch and `just recreate`. Test thumbnails and confirm they work.

If you have access to the Openverse secrets, you can add the `PHOTON_AUTH_KEY` variable to your `api/.env` file to enable query forwarding.

I am using the following additional sample data to be able to test the query forwarding:

```diff
diff --git a/sample_data/sample_images.csv b/sample_data/sample_images.csv
index 79a691a9..ecbb61c6 100644
--- a/sample_data/sample_images.csv
+++ b/sample_data/sample_images.csv
@@ -4999,3 +4999,4 @@ d4307b61-bd39-44bb-a3cd-aa760029d051,2021-12-15 20:48:57.914727+00,2021-12-15 20
 954a4dce-12b3-4323-86f3-8af7177635d6,2021-12-15 20:48:57.914727+00,2021-12-15 20:48:57.914727+00,provider_api,stocksnap,stocksnap,UCWLBKL1OL,https://stocksnap.io/photo/UCWLBKL1OL,https://cdn.stocksnap.io/img-thumbs/960w/UCWLBKL1OL.jpg,https://cdn.stocksnap.io/img-thumbs/280h/UCWLBKL1OL.jpg,2574,3218,1088013,cc0,1.0,Matt Bango,https://mattbango.photo,Music Guitar Photo,"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""downloads_raw"": ""5"", ""favorites_raw"": ""0"", ""page_views_raw"": ""225"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}","[{""name"": ""music"", ""provider"": ""stocksnap""}, {""name"": ""guitar"", ""provider"": ""stocksnap""}, {""name"": ""play"", ""provider"": ""stocksnap""}, {""name"": ""string"", ""provider"": ""stocksnap""}, {""name"": ""man"", ""provider"": ""stocksnap""}, {""name"": ""hand"", ""provider"": ""stocksnap""}, {""name"": ""musician"", ""provider"": ""stocksnap""}, {""name"": ""event"", ""provider"": ""stocksnap""}, {""name"": ""playing"", ""provider"": ""stocksnap""}, {""name"": ""instrument"", ""provider"": ""stocksnap""}, {""name"": ""band"", ""provider"": ""stocksnap""}, {""name"": ""electric"", ""provider"": ""stocksnap""}, {""name"": ""live"", ""provider"": ""stocksnap""}, {""name"": ""stage"", ""provider"": ""stocksnap""}, {""name"": ""sitting"", ""provider"": ""stocksnap""}, {""name"": ""person"", ""provider"": ""stocksnap""}, {""name"": ""player"", ""provider"": ""stocksnap""}, {""name"": ""group"", ""provider"": ""stocksnap""}, {""name"": ""performance"", ""provider"": ""stocksnap""}]",f,2021-12-15 20:48:57.914727+00,f,jpg,photograph,0.19767441860465113
 20c931c6-c0d6-4395-a2ee-506ad6d9dd69,2021-12-15 20:48:57.914727+00,2021-12-15 20:48:57.914727+00,provider_api,stocksnap,stocksnap,ZFJEKSUY76,https://stocksnap.io/photo/ZFJEKSUY76,https://cdn.stocksnap.io/img-thumbs/960w/ZFJEKSUY76.jpg,https://cdn.stocksnap.io/img-thumbs/280h/ZFJEKSUY76.jpg,3888,2592,472754,cc0,1.0,Foodie Girl,https://stocksnap.io/author/121423,Fresh Garlic Photo,"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""downloads_raw"": ""5"", ""favorites_raw"": ""0"", ""page_views_raw"": ""160"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}","[{""name"": ""fresh"", ""provider"": ""stocksnap""}, {""name"": ""garlic"", ""provider"": ""stocksnap""}, {""name"": ""ingredient"", ""provider"": ""stocksnap""}, {""name"": ""vegetable"", ""provider"": ""stocksnap""}, {""name"": ""raw"", ""provider"": ""stocksnap""}, {""name"": ""food"", ""provider"": ""stocksnap""}, {""name"": ""bulb"", ""provider"": ""stocksnap""}, {""name"": ""white"", ""provider"": ""stocksnap""}, {""name"": ""healthy"", ""provider"": ""stocksnap""}, {""name"": ""organic"", ""provider"": ""stocksnap""}, {""name"": ""plant"", ""provider"": ""stocksnap""}, {""name"": ""vegetarian"", ""provider"": ""stocksnap""}, {""name"": ""harvest"", ""provider"": ""stocksnap""}, {""name"": ""head"", ""provider"": ""stocksnap""}, {""name"": ""bunch"", ""provider"": ""stocksnap""}, {""name"": ""clove"", ""provider"": ""stocksnap""}, {""name"": ""green"", ""provider"": ""stocksnap""}, {""name"": ""nature"", ""provider"": ""stocksnap""}, {""name"": ""closeup"", ""provider"": ""stocksnap""}, {""name"": ""market"", ""provider"": ""stocksnap""}]",f,2021-12-15 20:48:57.914727+00,f,jpg,photograph,0.19767441860465113
 230c6951-7f83-4263-a0da-a151f9f67b24,2021-12-15 20:48:57.914727+00,2021-12-15 20:48:57.914727+00,provider_api,stocksnap,stocksnap,NLJESZLWIU,https://stocksnap.io/photo/NLJESZLWIU,https://cdn.stocksnap.io/img-thumbs/960w/NLJESZLWIU.jpg,https://cdn.stocksnap.io/img-thumbs/280h/NLJESZLWIU.jpg,4437,3328,950814,cc0,1.0,The Building Envelope,https://stocksnap.io/author/129953,Concrete Wall Photo,"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""downloads_raw"": ""5"", ""favorites_raw"": ""0"", ""page_views_raw"": ""210"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}","[{""name"": ""concrete"", ""provider"": ""stocksnap""}, {""name"": ""wall"", ""provider"": ""stocksnap""}, {""name"": ""building"", ""provider"": ""stocksnap""}, {""name"": ""architecture"", ""provider"": ""stocksnap""}, {""name"": ""abstract"", ""provider"": ""stocksnap""}, {""name"": ""background"", ""provider"": ""stocksnap""}, {""name"": ""city"", ""provider"": ""stocksnap""}, {""name"": ""simplicity"", ""provider"": ""stocksnap""}, {""name"": ""dirty"", ""provider"": ""stocksnap""}, {""name"": ""material"", ""provider"": ""stocksnap""}, {""name"": ""monochrome"", ""provider"": ""stocksnap""}, {""name"": ""pillars"", ""provider"": ""stocksnap""}, {""name"": ""pattern"", ""provider"": ""stocksnap""}, {""name"": ""texture"", ""provider"": ""stocksnap""}, {""name"": ""exterior"", ""provider"": ""stocksnap""}, {""name"": ""weathered"", ""provider"": ""stocksnap""}, {""name"": ""rough"", ""provider"": ""stocksnap""}, {""name"": ""surface"", ""provider"": ""stocksnap""}, {""name"": ""weathered"", ""provider"": ""stocksnap""}]",f,2021-12-15 20:48:57.914727+00,f,jpg,photograph,0.19767441860465113
+230c6951-7f83-4263-a0da-a151f9f67b21,2021-12-15 20:48:57.914727+00,2021-12-15 20:48:57.914727+00,provider_api,stocksnap,stocksnap,NLJESZLWIZ,https://stocksnap.io/photo/NLJESZLWIZ,https://images.nypl.org/index.php?id=489681&t=w&suffix=510d47dc-3a80-a3d9-e040-e00a18064a99.001,https://images.nypl.org/index.php?id=489681&t=w&suffix=510d47dc-3a80-a3d9-e040-e00a18064a99.001,4437,3328,950814,cc0,1.0,The Building Envelope,https://stocksnap.io/author/129953,Concrete Wall Photo,"{""license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/"", ""downloads_raw"": ""5"", ""favorites_raw"": ""0"", ""page_views_raw"": ""210"", ""raw_license_url"": ""https://creativecommons.org/publicdomain/zero/1.0/""}","[{""name"": ""concrete"", ""provider"": ""stocksnap""}, {""name"": ""wall"", ""provider"": ""stocksnap""}, {""name"": ""building"", ""provider"": ""stocksnap""}, {""name"": ""architecture"", ""provider"": ""stocksnap""}, {""name"": ""abstract"", ""provider"": ""stocksnap""}, {""name"": ""background"", ""provider"": ""stocksnap""}, {""name"": ""city"", ""provider"": ""stocksnap""}, {""name"": ""simplicity"", ""provider"": ""stocksnap""}, {""name"": ""dirty"", ""provider"": ""stocksnap""}, {""name"": ""material"", ""provider"": ""stocksnap""}, {""name"": ""monochrome"", ""provider"": ""stocksnap""}, {""name"": ""pillars"", ""provider"": ""stocksnap""}, {""name"": ""pattern"", ""provider"": ""stocksnap""}, {""name"": ""texture"", ""provider"": ""stocksnap""}, {""name"": ""exterior"", ""provider"": ""stocksnap""}, {""name"": ""weathered"", ""provider"": ""stocksnap""}, {""name"": ""rough"", ""provider"": ""stocksnap""}, {""name"": ""surface"", ""provider"": ""stocksnap""}, {""name"": ""weathered"", ""provider"": ""stocksnap""}]",f,2021-12-15 20:48:57.914727+00,f,jpg,photograph,0.19767441860465113
```

This is a quick and dirty way to get a URL with a query parameter into the sample data. Go directly to the image URL using the identifier: http://localhost:50280/v1/images/230c6951-7f83-4263-a0da-a151f9f67b21/

It should work fine. Now click the "thumbnail" link and see that the thumbnail is as expected (compressed, resized version of the upstream image).

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -82,6 +73,7 @@ services:
       - cache
     env_file:
       - api/env.docker
+      - api/.env
```
Comment (AetherUnbound) [inline docker-compose.yml:76]: Great idea! I like this for longer-lasting changes and I was always frustrated by having to jettison changes to `api/env.docker`.

**Review by AetherUnbound — APPROVED** [review]
This looks great! Thanks for the testing instructions, I was able to get the image with query params to work exactly as expected 😄 

I know you have a few more changes to make and I'm happy to re-review at that point if you'd like ✅ 🚀

File: docker-compose.yml (hunk 1 above)
Comment (sarayourfriend) [reply docker-compose.yml:76]: Also, risky! Too easy to accidentally commit secrets if I put them in that file :grimacing:

**Comment by sarayourfriend** [comment]
@obulat @dhruvkb Please review this PR when you have a chance. It will block further progress on the API ECS deployment work at a certain point.

**Comment by AetherUnbound** [comment]
@sarayourfriend I rebased this PR so it could include the fix from #1059 and build properly. It looks like the new `.env` file in the docker-compose is causing some issue in CI as well so I tried to add a fix for that, hope that's alright!

**Comment by sarayourfriend** [comment]
Thanks, Madison!

**Review by obulat — APPROVED** [review]
Sorry for late review. I tried running this PR yesterday, before Madison's fixes, and `just recreate` was failing.

Today I managed to run the steps in `testing`, and it worked fine. I tried to see what request we send to `photon` in the Docker logs, but couldn't find anything. All I see is this:
```
2023-01-06 16:38:49 [2023-01-06 13:38:49,838 - log_request_id.middleware -  47][INFO] [627312109a744e3f93e755da58f53582] method=GET path=/v1/images/230c6951-7f83-4263-a0da-a151f9f67b21/thumb/ status=200
2023-01-06 16:38:49 [06/Jan/2023 13:38:49] "GET /v1/images/230c6951-7f83-4263-a0da-a151f9f67b21/thumb/ HTTP/1.1" 200 49764
```

Do we intentionally not log the photon request, or is it not necessary?

**Comment by sarayourfriend** [comment]
@obulat We only logged thumbnail requests as debug logs, as far as I could tell. I'm not entirely sure what the usefulness of them is unless we have a specific issue we're trying to diagnose. I merged the PR as is because it'd been open for quite some time, but we can iterate in a separate PR to add additional logging if desired.

**Pull request merged**